activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/ActivitExpressionParser.java

```diff
@@ -0,0 +1,67 @@
+package org.activiti.runtime.api.impl;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.activiti.engine.delegate.DelegateExecution;
+import org.activiti.engine.delegate.Expression;
+import org.activiti.engine.impl.context.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ActivitExpressionParser {
+    
+    private final Logger logger = LoggerFactory.getLogger(ActivitExpressionParser.class);
+
+    private static final Pattern EL_PATTERN = Pattern.compile("([\\$]\\{([^\\}]*)\\})");
+    private static final int EL_KEY = 1;
+
+    private DelegateExecution execution;
+
+    public ActivitExpressionParser(DelegateExecution execution) {
+        this.execution = execution;
+    }
+
+    public Map<String, Object> resolveExpressionsMap(final Map<String, ? extends Object> sourceMap) {
+        final Map<String, Object> result = new LinkedHashMap<>();
+        sourceMap.forEach((key,
+                           value) -> result.put(key,
+                                                resolveExpressions(value)));
+        return result;
+    }
+
+    public String resolveExpressionsString(final String sourceString) {
+        if (sourceString == null || sourceString.isEmpty()) {
+            return sourceString;
+        }
+        final Matcher matcher = EL_PATTERN.matcher(sourceString);
+        final StringBuffer sb = new StringBuffer();
+        while (matcher.find()) {
+            final String expressionKey = matcher.group(EL_KEY);
+            final Expression expression = Context.getProcessEngineConfiguration().getExpressionManager().createExpression(expressionKey);
```

Comment:
This is the only point in your review I don't get. I looked for all the references in Activiti of this class and in every place I found it, they get it from the context. 

Do you really think it is worth to create the bean?

Reply:
The main problem with this requirement, as far as I understand, is that the `ExpressionManager` has some beans in its context (as you can see  [here](https://github.com/Activiti/Activiti/blob/db32ab086be8bc2fc958545ca1dae2ee80ddcdf4/activiti-engine/src/main/java/org/activiti/engine/impl/cfg/ProcessEngineConfigurationImpl.java#L1984)) when it is initialized. But maybe, I am missing something.

In my opinion, there are 2 options:

1. Using a supplier. 
This way, we can get set the `ExpressionManager` as a parameter in the  `ExpressionResolver` constructor, and use it later with its context once it is initialized. It would be something like:
```
    @Bean
    @ConditionalOnMissingBean
    public Supplier<ExpressionResolver> expressionResolverSupplier() {
        return () -> new ExpressionResolver(Context.getProcessEngineConfiguration().getExpressionManager());
    }
```

2. Creating a new instance of the `ExpressionManager` class without context as a Bean
```
    @Bean
    @ConditionalOnMissingBean
    public ExpressionManager expressionManager() {
        return new ExpressionManager();
    }
    
    @Bean
    @ConditionalOnMissingBean
    public ExpressionResolver expressionResolver() {
        return new ExpressionResolver(expressionManager());
    }
```

I can not figure out how to define the `ExpressionManager` as a bean without creating a new Instance of the `ExpressionManager` class with no context at all.

Any suggestions would be appreciated.

Reply:
Why don't you inject the expressionManager bean directly into expressionResolver?

    @Bean
    @ConditionalOnMissingBean
    public ExpressionManager expressionManager() {
        return new ExpressionManager();
    }
    
    @Bean
    @ConditionalOnMissingBean
    public ExpressionResolver expressionResolver(ExpressionManager expressionManager) {
        return new ExpressionResolver(expressionManager);
    }

Reply:
Chose 2nd option